daemon/prune.go

```diff
@@ -152,7 +152,9 @@ func (daemon *Daemon) VolumesPrune(ctx context.Context, pruneFilters filters.Arg
 		return nil
 	}
 
-	err = daemon.traverseLocalVolumes(pruneVols)
+	if err := daemon.traverseLocalVolumes(pruneVols); err == context.Canceled {
+		return rep, nil
```

Comment:
related to my comment two lines below; this doesn't return if `err == nil`,  correct?